### DISCUSSION TITLE
combined use with Composable Architecture's @Shared property

### DISCUSSION THREAD

**Discussion by ckse93** [pointfreeco/swift-sharing#9] [Q&A] [answered]
Hi, would it be possible to mix the usage with composable architecture's shared property?

**Comment by mbrandonw** [comment, marked as answer]
Hi @ckse93, can you clarify a bit what you mean?

Once [this](https://github.com/pointfreeco/swift-composable-architecture/pull/3517) PR is merged and released the TCA library will depend on this library for the `@Shared` tools. And so there will really only be one version of these tools out in the wild to use.

**Reply by ckse93** [reply]
considering the case where I have a `UIKitViewController`, and it has `@Sharing.Shared(.inMemory("count")) var count: Int = 1`. 
and in a different swiftUI View, it has the same `@Sharing.Shared(.inMemory("count")) var count: Int = 1`. 
and finally, in another place, I have a `TCAFeature` and within `TCAFeature.State` it has `@Shared(.inMemory("count")) var count: Int `.  
would all `count` variable point to the same location in memory, and would updating `count` animate the view?

**Reply by mbrandonw** [reply]
No, those would be different, but also there is no reason to use an old version of TCA with this library. Old versions of TCA already have all of the state sharing tools, so just use those.

Is there a reason you are trying to use an old version of TCA with this library?
